src/Karma/Application.php

```diff
@@ -119,10 +121,36 @@ private function initializeSourceFileSystem()
             {
                 $adapter->mount($path, new Local($path));
             }
-             
+
+            return $adapter;
+        };
+
+        $this['target.fileSystem.adapter'] = function($c) {
+
+            if($c['target.path'] !== $c['sources.path'])
+            {
+                $c['hydrator.allowNonDistFilesOverwrite'] = true;
+            }
+
+            if(! is_array($c['target.path']))
+            {
+                return new Local($c['target.path']);
+            }
+
+            $adapter = new MultipleAdapter();
+
+            foreach($c['target.path'] as $path)
+            {
+                $adapter->mount($path, new Local($path));
```

Comment:
The targetPath parameter can reference only a unique path.

It can be multiple if it is not set : in this case targetPath = sourcePath
If there is multiple targets it keeps the original behavior (files are generated on each config path)

There is a limited case : duplicated name of files on multiple sources. If targetPath is set, the files are generated on the root of target directory, and the file (the second one on hydrate pass) will overwrite the first. It could be possible to throw an exception or merge the files... to studying